nixos/modules/services/hardware/bluetooth.nix

```diff
@@ -30,6 +44,11 @@ in
 
     environment.systemPackages = [ bluez-bluetooth pkgs.openobex pkgs.obexftp ];
 
+    environment.etc = singleton {
+      source = pkgs.writeText "main.conf" cfg.extraConfig;
+      target = "bluetooth/main.conf";
+    };
```

Comment:
In general, we don't like polluting the `/etc` namespace with files that aren't needed. In this case it seems the config cannot be overridden for bluetoothd according to its manpage, so we'll have to do it like that.